zarr/core.py

```diff
@@ -2120,7 +2151,7 @@ def _encode_chunk(self, chunk):
             cdata = chunk
 
         # ensure in-memory data is immutable and easy to compare
-        if isinstance(self.chunk_store, MutableMapping):
+        if isinstance(self.chunk_store, KVStore):
```

Comment:
Since all store classes derived `BaseStore` also derives from `MutableMapping`, this check was always `True`.
Now by checking against `KVStore`, we actually check if the store is `in-memory`.